internal/storage/image.go

```diff
@@ -873,3 +883,43 @@ func GetImageService(ctx context.Context, sc *types.SystemContext, store storage
 
 	return is, nil
 }
+
+// FilterPinnedImage checks if the give image needs to be pinned
+// to exclude from kubelet's image GC.
+func FilterPinnedImage(image string, pinnedImages []*regexp.Regexp) bool {
+	if len(pinnedImages) == 0 {
+		return false
+	}
+
+	for _, pinnedImage := range pinnedImages {
+		if pinnedImage.MatchString(image) {
+			return true
+		}
+	}
+	return false
+}
+
+// CompileRegexpsForPinnedImages compiles regular expressions for the given
+// list of pinned images.
+func CompileRegexpsForPinnedImages(patterns *config.Config) []*regexp.Regexp {
```

Comment:
again, this is the server config. Since this only needs PinnedImages, maybe we should jsut pass that into this function instead of the whole config? will be easier to unit test that way